internal/cloud/testing.go

```diff
@@ -437,7 +437,7 @@ func testServerWithSnapshotsEnabled(t *testing.T, enabled bool) *httptest.Server
 			t.Fatal(err)
 		}
 
-		w.Header().Set("content-type", "application/json")
+		w.Header().Set("content-type", tfe.ContentTypeJSONAPI)
```

Comment:
Without that tests are failing for unknown content type - 
should be the same as https://github.com/opentofu/opentofu/blob/87de080aa206f9f6e8cad92de3fd68aa7c141460/internal/cloud/state.go#L588